### PR TITLE
GH-128: Move to Kafka-0.10 and Remove 0.8 support

### PR DESCRIPTION
**Pull request by artembilan**
Fixes GH-128 (https://github.com/spring-projects/spring-integration-java-dsl/issues/128)

* Remove `Kafka09` factory and all the stuff around
* Replace `Kafka` factory methods to use `Spec`s for Spring Integration Kafka 2.1.x adapters
* Rework `KafkaTests` to the SIK-2.1 style
* Remove SIK-1.x dependency
* Upgrade to SIK-2.1, therefore Spring Kafka 1.1 and Kafka-0.10
* Kafka 0.9 is still in support if the proper SIK dependency is used in the classpath

Tested with:
```
testRuntime ("org.springframework.integration:spring-integration-kafka:2.0.1.RELEASE") {
	force = true  // enforce 2.0.x
}
testRuntime ("org.springframework.kafka:spring-kafka-test:1.0.4.RELEASE") {
	force = true  // enforce 1.0.x
}
```

### CODE REVIEW

**Comment by artembilan** [comment]
@wilkinsona ,

let us know if this is more closer to IO requirements.

We decided to drop Kafka 0.8 support.
Have moved to Kafka 0.10 support directly, but Kafka 0.9 works well, if appropriate dependencies are in classpath.

Tested with:
```
gradlew clean springIoCheck -PplatformVersion=Brussels-BUILD-SNAPSHOT -PJDK8_HOME=d:/Java/jdk8
```
as well, BTW.

**Comment by wilkinsona** [comment]
Sounds good to me. Thank you

File: src/main/java/org/springframework/integration/dsl/kafka/Kafka.java [hunk 1]
```diff
@@ -16,18 +16,19 @@
...
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.AbstractMessageListenerContainer;
+import org.springframework.kafka.listener.config.ContainerProperties;
+import org.springframework.kafka.support.TopicPartitionInitialOffset;
 
 /**
- * Factory class for Kafka components.
+ * Factory class for Kafka-0.9, 0.10 components.
```
Comment (garyrussell) [inline src/main/java/org/springframework/integration/dsl/kafka/Kafka.java]: I don't think we should have versions here; changes might not be needed for 0.11

Comment (artembilan) [inline src/main/java/org/springframework/integration/dsl/kafka/Kafka.java]: Good observation!
Will fix soon with some other found flaws. :smile:

**Comment by garyrussell** [comment]
Merged as 07c210ed407e46d0992ee358a024cc283e8692ed

**Pull request closed without merging**